guide-app/src/components/shared/CalendarEvent.js

```diff
@@ -0,0 +1,156 @@
+// @flow
+import React from "react";
+import {
+  Image,
+  Linking,
+  StyleSheet,
+  Text,
+  TouchableOpacity,
+  View
+} from "react-native";
+import { XmlEntities } from "html-entities";
+
+import { Colors, TextStyles } from "@assets/styles";
+import { StyleSheetUtils } from "@utils";
+import { eventCalendarURL } from "@data/urls";
+import { DateUtils } from "@utils";
+
+const defaultImage = require("@assets/images/no-image-featured-image.png");
+
+const styles = StyleSheet.create({
+  item: {
+    marginBottom: 30,
+    padding: 4,
+    width: "50%"
+  },
+  listItemImage: {
+    height: 205,
+    aspectRatio: 205 / 175
+  },
+  imageWrapper: {
+    overflow: "hidden",
+    alignItems: "stretch"
+  },
+  listItemTextContainer: {
+    alignItems: "center"
+  },
+  listItemHoursContainer: {
+    alignItems: "center",
+    marginTop: -10,
+    marginBottom: 4
+  },
+  listItemHoursView: {
+    borderRadius: 4,
+    backgroundColor: Colors.white,
+    paddingTop: 5,
+    paddingBottom: 4,
+    paddingHorizontal: 15,
+    shadowRadius: 12,
+    shadowColor: "black",
+    shadowOpacity: 0.16
+  },
+  listItemHoursText: StyleSheetUtils.flatten([
+    TextStyles.defaultFontFamily,
+    {
+      fontSize: 13,
+      lineHeight: 13,
+      fontWeight: "500",
+      fontStyle: "normal",
+      color: Colors.black
+    }
+  ]),
+  listLocationText: StyleSheetUtils.flatten([
+    TextStyles.defaultFontFamily,
+    {
+      fontSize: 12,
+      lineHeight: 14,
+      fontWeight: "500",
+      letterSpacing: 0.75,
+      textTransform: "uppercase",
+      color: Colors.gray3,
+      padding: 5,
+      textAlign: "center"
+    }
+  ]),
+  listItemTitle: StyleSheetUtils.flatten([
+    TextStyles.defaultFontFamily,
+    {
+      fontSize: 16,
+      fontWeight: "900",
+      fontStyle: "normal",
+      color: Colors.black26,
+      paddingBottom: 8,
+      textAlign: "center",
+      width: "80%"
+    }
+  ]),
+  listItemDescription: StyleSheetUtils.flatten([
+    TextStyles.defaultFontFamily,
+    {
+      fontSize: 13,
+      lineHeight: 15,
+      letterSpacing: 0.19,
+      fontWeight: "normal",
+      fontStyle: "normal",
+      color: Colors.gray2C,
+      textAlign: "center",
+      paddingHorizontal: 4
+    }
+  ])
+});
+
+type Props = {
+  event: Event,
+  currentLanguage: string
+};
+
+function CalendarEvent({ event, currentLanguage }: Props) {
+  const {
+    description,
+    imageUrl,
+    location,
+    name,
+    slug,
+    dateStart,
+    dateEnd
+  } = event;
+  const image = imageUrl ? { uri: imageUrl } : defaultImage;
+
+  const entities = new XmlEntities();
+  const decodedLocationTitle = entities.decode(location.title);
```

Comment:
Do we want to do the decoding in this component, or move this out & elsewhere (like in a reducer?)? Seems like something more tied to the API than this particular component.

Reply:
Hm yeah. Is the reducer the best place you think?